llvm/tools/llvm-cas-object-format/llvm-cas-object-format.cpp

```diff
@@ -634,27 +644,44 @@ void StatCollector::printToOuts(ArrayRef<CASID> TopLevels) {
   llvm::sort(Kinds);
 
   size_t NumHashBytes = TopLevels.front().getHash().size();
-  outs() << "  => Note: 'Parents' counts incoming edges\n"
-         << "  => Note: 'Children' counts outgoing edges (to sub-objects)\n"
-         << "  => Note: HashSize = " << NumHashBytes << "B\n"
-         << "  => Note: PtrSize  = " << sizeof(void *) << "B\n"
-         << "  => Note: Cost     = Count*HashSize + PtrSize*Children + Data\n";
-  StringLiteral HeaderFormat = "{0,-22} {1,+10} {2,+7} {3,+10} {4,+7} {5,+10} "
-                               "{6,+7} {7,+10} {8,+7} {9,+10} {10,+7}\n";
-  StringLiteral Format = "{0,-22} {1,+10} {2,+7:P} {3,+10} {4,+7:P} {5,+10} "
-                         "{6,+7:P} {7,+10} {8,+7:P} {9,+10} {10,+7:P}\n";
-  outs() << llvm::formatv(HeaderFormat.begin(), "Kind", "Count", "", "Parents",
-                          "", "Children", "", "Data (B)", "", "Cost (B)", "");
-  outs() << llvm::formatv(HeaderFormat.begin(), "====", "=====", "",
-                          "=======", "", "========", "", "========", "",
-                          "========", "");
-  auto printInfo = [Format, NumHashBytes, &Totals = this->Totals](
-                       StringRef Kind, ObjectKindInfo Info) {
+
+  raw_fd_ostream &(*outStream)();
+  outStream = ComputeStatsCSV.empty() ? &outs : &outCSV;
+  if (ComputeStatsCSV.empty()) {
+    outStream()
+        << "  => Note: 'Parents' counts incoming edges\n"
+        << "  => Note: 'Children' counts outgoing edges (to sub-objects)\n"
+        << "  => Note: HashSize = " << NumHashBytes << "B\n"
+        << "  => Note: PtrSize  = " << sizeof(void *) << "B\n"
+        << "  => Note: Cost     = Count*HashSize + PtrSize*Children + Data\n";
+  }
+  StringLiteral HeaderFormatPretty =
+      "{0,-22} {1,+10} {2,+7} {3,+10} {4,+7} {5,+10} "
+      "{6,+7} {7,+10} {8,+7} {9,+10} {10,+7}\n";
+  StringLiteral FormatPretty =
+      "{0,-22} {1,+10} {2,+7:P} {3,+10} {4,+7:P} {5,+10} "
+      "{6,+7:P} {7,+10} {8,+7:P} {9,+10} {10,+7:P}\n";
+  StringLiteral FormatCSV = "{0}, {1}, {3}, {5}, {7}, {9}\n";
+
+  StringLiteral HeaderFormat =
+      ComputeStatsCSV.empty() ? HeaderFormatPretty : FormatCSV;
+  StringLiteral Format = ComputeStatsCSV.empty() ? FormatPretty : FormatCSV;
+
+  outStream() << llvm::formatv(HeaderFormat.begin(), "Kind", "Count", "",
```

Comment:
This might be fine but technically `StringLiteral` doesn't hold the null terminator but I guess there is always one.